.github/workflows/node-test.yml

```diff
@@ -0,0 +1,81 @@
+name: lint build test
+
+on: push
+
+permissions:
+  contents: read
+
+jobs:
+  lint-build-test:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v4
+      - name: Use Node.js 20.x
+        uses: actions/setup-node@v4
+        with:
+          node-version: '20'
+      - run: npm ci
+      - run: npm run lint
+      - run: npm run build
+      - run: npm test
+      - name: Upload coverage reports to Codecov
+        uses: codecov/codecov-action@v4.0.1
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
+          slug: fluidware-team/natuin
+  lint-dockerfile:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v4
+      - name: Lint Dockerfile with Hadolint
+        uses: hadolint/hadolint-action@v3.1.0
+  lint-editorconfig:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Lint with editorconfig-checker
+        uses: editorconfig-checker/action-editorconfig-checker@v2.0.0
+  oci-build-and-publish:
+    needs:
+      - lint-build-test
+      - lint-dockerfile
+      - lint-editorconfig
+    runs-on: ubuntu-22.04
+    permissions:
+      contents: read
+      packages: write
+    steps:
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+      - name: Login to Docker Hub
```

Comment:
I'd condition publishing to docker hub too

Reply:
It is indeed conditioned to the `ref` being a `tag`. Do you mean we also want a `PUBLISH_DH` flag?

Reply:
yes sorry, I meant in the same way we already do for GHC